bgpd/bgp_community.c

```diff
@@ -910,3 +910,142 @@ void community_finish(void)
 	hash_free(comhash);
 	comhash = NULL;
 }
+
+static unsigned int bgp_aggr_community_hash_key(void *data)
+{
+	struct community *community = data;
+
+	return jhash(community->val, community->size, 0xdcba4321);
+}
+
+static bool bgp_aggr_community_hash_cmp(const void *n1, const void *n2)
+{
+	const struct community *com1 = n1, *com2 = n2;
+
+	if (com1 == NULL && com2 == NULL)
```

Comment:
how are com1 or com2 ever null?  I don't see this as possible.

Reply:
Yes. It is not possible for com1 and com2 to both be NULL at the same time. I had added the additional checks as a guard against misuse of the API.
It makes sense to remove it.